Common/src/gRPC/Services/Submitter.cs

```diff
@@ -569,82 +568,61 @@ Task<IEnumerable<TaskStatusCount>> CountUpdateFunc()
-      {
-        return new AvailabilityReply
-               {
-                 Error = new TaskError
-                         {
-                           TaskId = ownerId,
-                         },
-               };
-      }
+      var result = await resultTable_.GetResult(request.Session,
+                                                request.Key,
+                                                contextCancellationToken)
+                                     .ConfigureAwait(false);
 
-      if (completion.Values.Any(count => count.Status is TaskStatus.Canceled or TaskStatus.Canceling))
+      switch (result.Status)
       {
-        return new AvailabilityReply
-               {
-                 NotCompletedTask = ownerId,
-               };
+        case ResultStatus.Completed:
+          return new AvailabilityReply
+                 {
+                   Ok = new Empty(),
+                 };
+        case ResultStatus.Created:
+          break;
+        case ResultStatus.Aborted:
+          var taskData = await taskTable_.ReadTaskAsync(result.OwnerTaskId,
+                                                        contextCancellationToken)
+                                         .ConfigureAwait(false);
+          
+          return new AvailabilityReply
+                 {
+                   Error = new TaskError
+                           {
+                             TaskId = taskData.TaskId,
+                             Error =
+                             {
+                               new Error
+                               {
+                                 Detail     = taskData.Output.Error,
+                                 TaskStatus = taskData.Status,
+                               },
+                             },
+                           },
+                 };
+        case ResultStatus.Unspecified:
+        default:
+          throw new ArgumentOutOfRangeException();
       }
 
-      result = await resultTable_.GetResult(request.Session,
-                                            request.Key,
-                                            contextCancellationToken)
-                                 .ConfigureAwait(false);
-      logger_.LogDebug("OwnerTaskId {OwnerTaskId}",
-                       result.OwnerTaskId);
-      if (ownerId != result.OwnerTaskId)
-      {
-        continueWaiting = result.Status != ResultStatus.Completed;
-        if (continueWaiting)
-        {
-          await Task.Delay(150,
-                           contextCancellationToken)
-                    .ConfigureAwait(false);
-        }
-      }
-      else
+      await Task.Delay(currentPollingDelay,
+                       contextCancellationToken)
+                .ConfigureAwait(false);
+      if (2 * currentPollingDelay < taskTable_.PollingDelayMax)
       {
-        continueWaiting = false;
+        currentPollingDelay = 2 * currentPollingDelay;
```

Comment:
where the magic factor 2 comes from?

Reply:
We increase the delay by two until we reach the allocated maximum